src/main/java/com/iexec/core/task/TaskService.java

```diff
@@ -432,6 +433,63 @@ private void initializedOrRunning2ContributionTimeout(Task task) {
         }
     }
 
+    /**
+     * When all workers have failed to run the task, we send the new status as soon as possible.
+     * It avoids a state within which we know the task won't be updated anymore, but we wait for a timeout.
+     * <br>
+     * We consider that all workers are in a `RUNNING_FAILED` status when:
+     * <ol>
+     *     <li>All alive workers have tried to run the task;</li>
+     *     <li>All alive workers that have tried to run the task have failed.</li>
+     * </ol>
+     *
+     * @param task Task to check and to make become {@link TaskStatus#RUNNING_FAILED}.
+     */
+    private void running2RunningFailed(Task task) {
+        boolean isRunningTask = task.getCurrentStatus().equals(RUNNING);
+        if (!isRunningTask) {
+            return;
+        }
+
+        final ReplicatesList replicatesList = replicatesService.getReplicatesList(task.getChainTaskId()).orElseThrow();
+        final List<Worker> aliveWorkers = workerService.getAliveWorkers();
+
+        final List<Replicate> replicatesOfAliveWorkers = aliveWorkers
+                .stream()
+                .map(Worker::getWalletAddress)
+                .map(replicatesList::getReplicateOfWorker)
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(Collectors.toList());
+
+        // If at least an alive worker has not run the task, it is not a `RUNNING_FAILURE`.
+        final boolean notAllAliveWorkersTried = replicatesOfAliveWorkers.size() != aliveWorkers.size();
+
+        if (notAllAliveWorkersTried) {
+            return;
+        }
+
+        // If not all alive workers have failed while running the task, that's not a running failure.
+        boolean notAllReplicatesFailed = replicatesOfAliveWorkers
+                .stream()
+                .map(Replicate::getLastRelevantStatus)
+                .map(Optional::get)
+                .anyMatch(Predicate.not(ReplicateStatus::isFailedBeforeComputed));
```

Comment:
Detail: don't you think the positive way is easier to read? (same for above)

Reply:
It's easier, but could be a bit slower depending on the workerpool size.
However, for the above test, it wouldn't change anything in time complexity to revert the condition and the variable name. I'll do so.

Reply:
I agree, thanks